v0/util/index.js

```diff
@@ -1020,6 +1020,25 @@ const getDestinationExternalIDInfoForRetl = (message, destination) => {
   return { destinationExternalId, objectType, identifierType };
 };
 
+const getDestinationExternalIDObject = (message, destination) => {
+  let externalIdArray = [];
+  if (message.context && message.context.externalId) {
+    externalIdArray = message.context.externalId;
+  }
+  let obj;
+  if (externalIdArray) {
+    externalIdArray.forEach(extIdObj => {
+      const { type } = extIdObj;
+      if (type.includes(`${destination}-`)) {
+        obj = extIdObj;
+        return;
```

Comment:
we are assigning the found `extIdObj` to `obj` variable however not utilizing it. I think we have an unexpected return statement after the assignment. We can use some() for looking up values and break out when the element is found.